src/Management/ManagementClient.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Doofinder\Management;
+
+use Doofinder\Configuration;
+use Doofinder\Management\Resources\Indexes;
+use Doofinder\Management\Resources\Items;
+use Doofinder\Management\Resources\SearchEngines;
+use Doofinder\Shared\HttpClient;
+
+class ManagementClient
+{
+    private $searchEnginesResource;
+    private $itemsResource;
+    private $indexesResource;
+
+    private function __construct($host, $token)
+    {
+        $config = Configuration::create($host, $token);
```

Comment:
I think you don't need the `Configuration` class, as it is only useful to handle only two params, that the `ManagementClient` (or `SearchClient`) can have, it could be a dictionary (key value array) as follow

```
$array = array(
    "host" => "https://eu1-api.doofinder.com",
    "token" => "...",
);
```

Reply:
Yes, at the moment we have two parameters but we will need more to configure curl, for example.
Anyway I prefer to have a separated config, but is just my opinion. Do you prefer to have it inside the Mangement class?

Reply:
I Agree with @GLagunilla at the moment we have only two variables, but we could have more as curl options or something more elaborated.